### ISSUE TITLE
Conflicting SQL relationships

### ISSUE THREAD

**Issue by pastewka** [jic-dtool/dtool-lookup-server#20]
I am getting the following warning, only tested with latest version 0.17.1
```
/usr/local/lib/python3.8/site-packages/flask_sqlalchemy/__init__.py:550: SAWarning: relationship 'BaseURI.register_users' will copy column user.id to column register_permissions.user_id, which conflicts with relationship(s): 'User.register_base_uris' (copies user.id to register_permissions.user_id). If this is not the intention, consider if these relationships should be linked with back_populates, or if viewonly=True should be applied to one or more if they are read-only. For the less common case that foreign key constraints are partially overlapping, the orm.foreign() annotation can be used to isolate the columns that should be written towards.   The 'overlaps' parameter may be used to remove this warning.
```

**Comment by tjelvar-olsson** [comment]
Hi Lars,

I don't seem to be able to reproduce this (when running the test suite).

What version of SQLAlchemy are you using?
What steps do you go through before you see this warning message?

Best,
Tjelvar

**Comment by pastewka** [comment]
This occurs in a production environment during exection of `flask base_uri index ecs://frct-simdata` with:
```
Flask-SQLAlchemy                            2.5.1
SQLAlchemy                                  1.4.15
```

**Comment by tjelvar-olsson** [comment]
Thanks I can reproduce it now.

**Comment by tjelvar-olsson** [comment]
This should have been fixed in 0.17.2 release.

**Comment by pastewka** [comment]
I can confirm that the warning has disappears, thanks @tjelvar-olsson
